### PR TITLE
[DEVX-217] Cleanup Repo

### PR DESCRIPTION
**Pull request by sainivedh**
## What
 - Includes removal and refactor of long pending items for the repo

### Delete
   - `clarifai.auth` as now langchain is updated
   -  Dataset Upload `Examples` and `Loaders` which is available in clarifai-examples repo
   - Remove `split` arg in `upload_dataset()`
   - Remove `task` arg in `upload_dataset()`
   - Remove `module_dir` in `upload_dataset`
  
### Refactor
 - Rename `chunk_size` to `batch_size` in dataset/inputs
 - Rename `url_init` arg name to  url
 - Refactor `ClarifaiDataLoader` to include `task` property
 - Refactor `classes` to `labels` in all features to be consistent
 - Support multi-labels classification in dataset upload
 - Add `DATASET_UPLOAD_TYPES` in constants
 -  Restructure `dataset_loader` to more usable `dataloader` in `upload_dataset`
 - Set default `batch_size` to 32 from 128
 - Change `text_cls` task name to full `text_classification`
 - Modified `loaders`
 - Review comments

### CODE REVIEW

File: clarifai/client/dataset.py [hunk 1]
```diff
@@ -282,46 +284,44 @@ def _data_upload(self, dataset_obj: ClarifaiDatasetType) -> None:
...
+                     batch_size: int = 32) -> None:
     """Uploads a dataset to the app.
 
     Args:
-      task (str): task type(text_clf, visual-classification, visual_detection, visual_segmentation, visual-captioning)
-      split (str): split type(train, test, val)
       module_dir (str): path to the module directory
-      dataset_loader (str): name of the dataset loader
-      chunk_size (int): chunk size for concurrent upload of inputs and annotations
+      dataloader (Type[ClarifaiDataLoader]): ClarifaiDataLoader object
```
Comment (isaac-chung) [inline clarifai/client/dataset.py:294]: Nit: Two separate words, so maybe should keep snake case.
```suggestion
      data_loader (Type[ClarifaiDataLoader]): ClarifaiDataLoader object
```

File: clarifai/datasets/upload/base.py [hunk 2]
```diff
@@ -53,9 +52,15 @@ def get_protos(self, input_ids: List[int]
...
+  def __init__(self) -> None:
     pass
 
+  @property
+  def task(self):
+    raise NotImplementedError("Task should be one of \
+                              'text_classification', 'visual-classification', \
+                              'visual_detection', 'visual_segmentation', 'visual-captioning'")
+
   def load_data(self) -> None:
```
Comment (isaac-chung) [inline clarifai/datasets/upload/base.py:64]: Use underscore in all?
Instead of hardcoding, could reference `DATASET_UPLOAD_TASKS`

File: clarifai/datasets/upload/image.py [hunk 3]
```diff
@@ -10,8 +10,8 @@
 
 class VisualClassificationDataset(ClarifaiDataset):
 
-  def __init__(self, datagen_object: Iterator, dataset_id: str, split: str) -> None:
-    super().__init__(datagen_object, dataset_id, split)
+  def __init__(self, datagen_object: Iterator, dataset_id: str) -> None:
```
Comment (isaac-chung) [inline clarifai/datasets/upload/image.py:13]: Same for all references.
```suggestion
  def __init__(self, data_generator: Iterator, dataset_id: str) -> None:
```

File: clarifai/datasets/upload/image.py [hunk 4]
```diff
@@ -28,22 +28,22 @@ def process_datagen_item(id):
       datagen_item = self.datagen_object[id]
```
Comment (isaac-chung) [inline clarifai/datasets/upload/image.py:28]:
```suggestion
      data_item = self.datagen_object[id]
```

File: clarifai/datasets/upload/loaders/coco_captions.py [hunk 5]
```diff
@@ -1,103 +0,0 @@
...
-import requests
-from pycocotools.coco import COCO
-from tqdm import tqdm
-
-from clarifai.datasets.upload.base import ClarifaiDataLoader
-
-from ..features import VisualClassificationFeatures
-
-
-class COCOCaptionsDataLoader(ClarifaiDataLoader):
```
Comment (isaac-chung) [inline clarifai/datasets/upload/loaders/coco_captions.py:16]: Curious, how come we are removing all these?

File: clarifai/datasets/upload/examples/README.md [hunk 6]
```diff
@@ -1,31 +0,0 @@
-## Dataset upload from local directory
```
Comment (isaac-chung) [inline clarifai/datasets/upload/examples/README.md:1]: Curious, how come we are removing all these?

**Review by isaac-chung — COMMENTED** [review]
Looks good. A few minor comments and questions.

File: clarifai/client/dataset.py (hunk 1 above)
Comment (sainivedh) [reply clarifai/client/dataset.py:294]: I think `dataloader` is the usual convention, pytorch etc, follows the same

File: clarifai/datasets/upload/examples/README.md (hunk 6 above)
Comment (sainivedh) [reply clarifai/datasets/upload/examples/README.md:1]: Examples are shifted to https://github.com/Clarifai/examples. So that it doesn't make Pypi build big with images/annotation files

File: clarifai/datasets/upload/image.py (hunk 4 above)
Comment (sainivedh) [reply clarifai/datasets/upload/image.py]: Done

File: clarifai/datasets/upload/base.py (hunk 2 above)
Comment (sainivedh) [reply clarifai/datasets/upload/base.py]: Done

File: clarifai/client/app.py [hunk 7]
```diff
@@ -28,24 +28,24 @@ class App(Lister, BaseClient):
   """App is a class that provides access to Clarifai API endpoints related to App information."""
 
   def __init__(self,
-               url_init: str = "",
+               app_url: str = "",
```
[the remark this answers is not in the document]
Comment (sainivedh) [reply clarifai/client/app.py]: Done

File: clarifai/datasets/upload/loaders/README.md [hunk 8]
```diff
@@ -4,43 +4,46 @@ A collection of data preprocessing modules for popular public datasets to allow
...
- |[xVIEW](http://xviewdataset.org/)  | Detection | `xview_detection` | `train`
- | [ImageNet](https://www.image-net.org/)  | Classification | `imagenet_classification` | `train`
+ | dataset name | task | module name (.py)
+ | --- | --- | ---
+ | [COCO 2017](https://cocodataset.org/#download) | Detection | `coco_detection` |
+ |        | Segmentation | `coco_segmentation` |
+ |       | Captions | `coco_captions` |
+ |[xVIEW](http://xviewdataset.org/)  | Detection | `xview_detection` |
+ | [ImageNet](https://www.image-net.org/)  | Classification | `imagenet_classification` |
 ## Contributing Modules
```
Comment (isaac-chung) [inline clarifai/datasets/upload/loaders/README.md:31]: Suggestion: To avoid confusion with Clarifai's Modules, maybe we could use the name of the directory.
```suggestion
## Contributing To Loaders
```

File: clarifai/datasets/upload/loaders/README.md [hunk 9]
```diff
@@ -4,43 +4,46 @@ A collection of data preprocessing modules for popular public datasets to allow
...
- | [COCO 2017](https://cocodataset.org/#download) | Detection | `coco_detection` | `train`, `val` |
- |        | Segmentation | `coco_segmentation` | `train`, `val` |
- |       | Captions | `coco_captions` | `train`, `val` |
- |[xVIEW](http://xviewdataset.org/)  | Detection | `xview_detection` | `train`
- | [ImageNet](https://www.image-net.org/)  | Classification | `imagenet_classification` | `train`
+ | dataset name | task | module name (.py)
+ | --- | --- | ---
+ | [COCO 2017](https://cocodataset.org/#download) | Detection | `coco_detection` |
+ |        | Segmentation | `coco_segmentation` |
+ |       | Captions | `coco_captions` |
```
Comment (isaac-chung) [inline clarifai/datasets/upload/loaders/README.md:28]: If these are not yet supported, maybe we can open up issues in the repo with "help wanted" / "good first issue" tag if the team does not plan on taking this up in the near future.

File: clarifai/datasets/upload/loaders/coco_detection.py [hunk 10]
```diff
@@ -1,134 +1,75 @@
...
-      else:  # if no annotations for given image_id-cat_id pair
+    concept_ids = []
+
+    input_ann_ids = self.coco.getAnnIds(imgIds=[value['id']])
+    input_anns = self.coco.loadAnns(input_ann_ids)
+
+    for ann in input_anns:
+      # get concept info
+      # note1: concept_name can be human readable
+      # note2: concept_id can only be alphanumeric, up to 32 characters, with no special chars except `-` and `_`
```
Comment (isaac-chung) [inline clarifai/datasets/upload/loaders/coco_detection.py:46]: Nit: No need to enumerate notes.

File: clarifai/datasets/upload/loaders/coco_segmentation.py [hunk 11]
```diff
@@ -2,165 +2,94 @@
...
+        poly[:, 0], poly[:, 1] = poly[:, 0] / value['width'], poly[:, 1] / value['height']
+        concept_ids.append(concept_id)
+      else:  # seg: {"counts":[...]}
+        if isinstance(ann['segmentation']['counts'], list):
+          rle = maskUtils.frPyObjects([ann['segmentation']], value['height'], value['width'])
+        else:
+          rle = ann['segmentation']
+        mask = maskUtils.decode(rle)  #binary mask
+        #convert mask to polygons and add to annots
+        contours, _ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
```
Comment (isaac-chung) [inline clarifai/datasets/upload/loaders/coco_segmentation.py:73]: Just found this, and is out of scope of this PR. Annotations support masks as base64 images: Annotation -> Data -> Image. How come we are converting them into polygons?

**Review by isaac-chung — APPROVED** [review]
Looks good. A few minor comments.

File: clarifai/datasets/upload/loaders/coco_segmentation.py (hunk 11 above)
Comment (sainivedh) [reply clarifai/datasets/upload/loaders/coco_segmentation.py:73]: For segmentation, only polygons are supported in dataloader
https://github.com/Clarifai/clarifai-python/blob/master/clarifai/datasets/upload/features.py

File: clarifai/client/app.py [hunk 12]
```diff
@@ -28,24 +28,24 @@ class App(Lister, BaseClient):
   """App is a class that provides access to Clarifai API endpoints related to App information."""
 
   def __init__(self,
-               url_init: str = "",
+               url: str = "",
                app_id: str = "",
                base_url: str = "https://api.clarifai.com",
```
[the remark this answers is not in the document]
Comment (sainivedh) [reply clarifai/client/app.py:33]: I assume you're referring to base_url?

File: clarifai/datasets/upload/loaders/README.md (hunk 9 above)
Comment (sainivedh) [reply clarifai/datasets/upload/loaders/README.md:28]: Good idea. But for now these are supported

File: clarifai/client/app.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (isaac-chung) [reply clarifai/client/app.py:33]: Yes

File: clarifai/datasets/upload/loaders/README.md (hunk 9 above)
Comment (isaac-chung) [reply clarifai/datasets/upload/loaders/README.md:28]: I see. I got confused as the cells for seg/captions were empty under "dataset name". Is it possible to merge those cells together for those 3 rows for that column?

File: clarifai/datasets/upload/loaders/coco_segmentation.py (hunk 11 above)
Comment (isaac-chung) [reply clarifai/datasets/upload/loaders/coco_segmentation.py:73]: Why is that, when we support seg masks?

File: clarifai/client/app.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (sainivedh) [reply clarifai/client/app.py:33]: Will change this in a later PR

File: clarifai/datasets/upload/loaders/README.md (hunk 9 above)
Comment (sainivedh) [reply clarifai/datasets/upload/loaders/README.md:28]: I don't think we can merge cells using plain markdown. We'll have to use custom html tags

Comment (isaac-chung) [reply clarifai/datasets/upload/loaders/README.md:28]: That's too bad. Let's leave it then.

File: clarifai/datasets/upload/loaders/coco_segmentation.py (hunk 11 above)
Comment (sainivedh) [reply clarifai/datasets/upload/loaders/coco_segmentation.py:73]: This is in todo planning, Need to extend the support for mask uploads.
